docs/source/user_guide/pipeline-components.md

```diff
@@ -0,0 +1,68 @@
+
+![A basic Kubeflow pipeline](../images/user_guide/pipeline-components/kubeflow-pipeline.png)
+
+The same pipeline could be implemented using a single component that performs all these tasks, but that component might not be as universally re-usable. Consider, for example, that for another project the data resides in a different kind of storage. With fine-granular components you'd only have to replace the load data component with one that supports the other storage type and could retain everything else. 
+
+Elyra includes three _generic components_ that allow for the processing of Jupyter notebooks, Python scripts, and R scripts. These components are called generic because they can be used in all runtime environments that Elyra pipelines currently support: local/JupyterLab, Kubeflow Pipelines, and Apache Airflow.
+
+![Generic components in the palette](../images/user_guide/pipeline-components/generic-components-in-palette.png)
+
+Note: Refer to the [_Best practices_ topic in the _User Guide_](best-practices-file-based-nodes.md) to learn more about special considerations for generic components.
+
+_Custom components_ are commonly only implemented for Kubeflow Pipelines or Apache Airflow, but not both.
+
+![Kubeflow components in the palette](../images/user_guide/pipeline-components/custom-kubeflow-components-in-palette.png)
+
+There are many example custom components available that you can utilize in pipelines, but you can also create your own. Details on how to create a component can be found in the [Kubeflow Pipelines documentation](https://www.kubeflow.org/docs/components/pipelines/sdk/component-development/) and the [Apache Airflow documentation](https://airflow.apache.org/docs/apache-airflow/stable/concepts/operators.html). Do note that in Apache Airflow components are called operators, but for the sake of consistency the Elyra documentation refers to them as components.
+
+#### Example custom components
+
+For illustrative purposes the Elyra component registry includes a few custom components that you can use to get started. 
+
+![Example pipeline for the HTTP component](../images/user_guide/pipeline-components/example-components-pipeline.png)
+
+Component details and demo pipelines can be found in the `https://github.com/elyra-ai/examples` repository:
+- [Kubeflow Pipeline components](https://github.com/elyra-ai/examples/tree/master/pipelines/kubeflow_pipelines_component_examples)
+- [Apache Airflow components](https://github.com/elyra-ai/examples/tree/master/pipelines/airflow_component_examples)
+
+Note that example components are provided as is. Unless indicated otherwise they are not maintained by the Elyra community.
+
+### Managing custom components
+
+In this release Elyra utilizes an internal registry to store information about custom components that you can use in the Visual Pipeline Editor to create pipelines.
+
+#### Adding a component to the registry
+
+This feature is currently not available.
```

Comment:
not available....yet?

Reply:
Wouldn't that be duplicate since _currently_ somewhat implies that  it might be there in the future.

Reply:
word play is fun